src/Plutarch/Test/Export.hs

```diff
@@ -0,0 +1,728 @@
+
+ @since 1.2.1
+-}
+exportTests ::
+  forall (dat :: Type) (red :: Type) (info :: Type) (param :: Type).
+  String ->
+  RawScriptExport ->
+  Linker param (ScriptExport info) ->
+  TestParams dat red param ->
+  WithExport dat red info () ->
+  TestTree
+exportTests name raw linker args comp = case runWithExport comp raw linker args of
+  Left err -> singleTest "Linking" . FailedToLink $ err
+  Right result -> testGroup name . toList . Map.foldMapWithKey go $ result
+    where
+      go :: Text -> PreparedScriptTests -> Acc TestTree
+      go testName = \case
+        NoSuchScript -> pure . singleTest "Presence" . MissingScript $ testName
+        PreparedTests tests -> fmap prepDataToTest tests
+
+{- | States that the named validator should have the given outcome, using the
+ currently in-scope arguments. The test will be labelled with the given
+ explanation (or reason) in the suite, as long as the named script actually
+ exists.
+
+ = Example
+
+ > tests = do
+ >    assertValidator Crashes "reason for crash" "myScript"
+ >    assertValidator Runs "reason for success" "aBetterScript"
+
+ @since 1.2.1
+-}
+assertValidator ::
+  forall (dat :: Type) (red :: Type) (info :: Type).
+  (ToData dat, ToData red) =>
+  -- | What you expect to happen
+  ScriptOutcome ->
+  -- | The reason or explanation to label the test with
+  String ->
+  -- | The name of the script to run
+  Text ->
+  WithExport dat red info ()
+assertValidator outcome reason name = WithExport $ do
+  lookedUp <- asks (view (#raws % #rawScripts % at' @(Map _ _) name))
+  case lookedUp of
+    Nothing -> tell . noSuchScript $ name
+    Just envelope -> case view (to tsRole) envelope of
+      ValidatorRole -> do
+        datum' <- asks (toData . view #datum)
+        redeemer' <- asks (toData . view #redeemer)
+        sc' <- asks (toData . view #context)
+        let script = view (to tsScript) envelope
+        let result = AppliedValidator script datum' redeemer' sc'
+        tell . prepareTest name reason outcome $ result
+      MintingPolicyRole -> tell . roleMismatch name reason $ ValidatorRole
+
+{- | As 'assertValidator', but for a minting policy instead.
+
+ @since 1.2.1
+-}
+assertMintingPolicy ::
+  forall (dat :: Type) (red :: Type) (info :: Type).
+  (ToData dat) =>
+  -- | What you expect to happen
+  ScriptOutcome ->
+  -- | The reason or explanation to label the test with
+  String ->
+  -- | The name of the script to run
+  Text ->
+  WithExport dat red info ()
+assertMintingPolicy outcome reason name = WithExport $ do
+  lookedUp <- asks (view (#raws % #rawScripts % at' @(Map _ _) name))
+  case lookedUp of
+    Nothing -> tell . noSuchScript $ name
+    Just envelope -> case tsRole envelope of
+      ValidatorRole -> tell . roleMismatch name reason $ MintingPolicyRole
+      MintingPolicyRole -> do
+        datum' <- asks (toData . view #datum)
+        sc' <- asks (toData . view #context)
+        let script = tsScript envelope
```

Comment:
CRITICAL!!!!

This should not use raw script. We need to look up linked script